### PR TITLE
Blog app Milestone 7 : Add Forms

### PR DESCRIPTION
**Pull request by NabilHY**
Hello, Thanks for offering your feedback on this PR
In this branch I have :
> Created a method called current_user in the ApplicationController to make current user data available to all controllers and returns the first user from the database.
> Create new forms for the Post and Comments. on behalf of the current_user using the method created in the ApplicationController.
> Allowed Users to add likes to Posts by a form that increments the likes_counter.

### CODE REVIEW

File: app/views/posts/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <%= form.text_area :text, class:"form-control", placeholder:"Text", rows: 4, required: true %>
+  </div>
+  <br />
+<<<<<<< HEAD
+  <div class="d-flex justify-content-center">
+     <%= form.submit "Post", class:"btn btn-secondary" %>
+=======
+ <div class="d-flex justify-content-center">
+    <%= form.submit "Post", class:"btn btn-secondary" %>
+>>>>>>> c38d85248c21e753f3d214779742cc222469800d
```
Comment (zainsadaqat) [inline app/views/posts/_form.html.erb]: Please remove ```<<<<< HEAD ====== >>>>>>``` this after resolving the conflicts to make the code look clean 👍🏻

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -2,10 +2,18 @@
...
+  <button type="button" class = "btn btn-dark mb-2 mt-2">
+      <%= link_to 'New Post', new_user_post_path %>
+  </button>
+<<<<<<< HEAD
+</div>
+<div class="d-flex justify-content-center">
+        <button type="button" class = "btn btn-dark mb-2 mt-2">
+            Pagination
+        </button>
+=======
```
Comment (zainsadaqat) [inline app/views/posts/index.html.erb]: Please remove this after resolving the conflicts to make the code look clean 👍🏻

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# Required Changes 🔁
# Hello @NabilHY  😄 

You're doing a great job on your project. We're proud of you. Good job so far! 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
  ✅  Correct use of Github Flow
  ✅  No linter errors
  ✅  Descriptive Pull Request

## Required Changes ♻️

_Check the comments under the review._




## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 2 above)
Comment (NabilHY) [reply app/views/posts/index.html.erb]: Thanks for bringing that up, I fixed the issue concerning conflicts

**Review by FrederickMih — APPROVED** [review]
# Status: Approved :tada: 

Hi @NabilHY, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Great PR title and summary :+1: 
- No Linter error :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**